src/app/copy/block-copy/block-copy.component.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  ViewEncapsulation
+} from '@angular/core';
+import { uniqueId } from 'lodash';
+
+import { CopyBase } from '../copy-base';
+
+import { CopyService } from '../copy-service/copy.service';
+
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'pfng-block-copy',
+  templateUrl: './block-copy.component.html',
+  styleUrls: ['./block-copy.component.less']
+})
+
+export class BlockCopyComponent extends CopyBase {
+  @Input('label') label: string;
+  @Input('expandBtnAriaLabel') expandBtnAriaLabel: string;
+  @Input('buttonLabel') buttonLabel: string = 'Copy';
+  @Input('expanded') expanded: boolean = false;
+
+  public uniqueID: string = uniqueId('pfng-block-copy-button-');
+
+  constructor(protected copyService: CopyService) {
+    super(copyService);
+  }
+
+  /**
+   * Used to uniquly relate label to copy button
+   */
+  get copyBtnId(): string {
+    return this.uniqueID;
```

Comment:
Then, this can return `this.uniqueID + '-button';`

The resulting ID would look more like `pfng-block-copy1-button`. And anywhere else we need an ID later, we could create something like `pfng-block-copy1-arialabel`, for example. 

This makes testing easier if each ID is prefixed similarly. This would also follow the convention in pfng-list.